plugins/ssr/ssrequality.ml

```diff
@@ -417,7 +417,7 @@ let rwcltac cl rdx dir sr gl =
       then errorstrm Pp.(str "Rewriting impacts evars")
       else errorstrm Pp.(str "Dependent type error in rewrite of "
         ++ pr_constr_env (pf_env gl) (project gl) (Term.mkNamedLambda pattern_id (EConstr.Unsafe.to_constr rdxt) (EConstr.Unsafe.to_constr cl)))
-    | CErrors.UserError _ as e -> raise e
+    | (CErrors.UserError _ | Pretype_errors.PretypeError _) as e -> raise e
     | e -> anomaly ("cvtac's exception: " ^ Printexc.to_string e);
```

Comment:
What is this trying to catch? Why not just let the exception bubble up?

Reply:
I don't know what I was trying to catch here. IMO it is my (bad?) habit of having assertions that backfires me (as the "no safe_id attached" one). `cvtac` is supposed to be a tactic, hence, IMO, it should only fail with `CErrors.UserError`. On purpose I type-check the inferred predicate and raise a different exception (few lines above) to re-raise a better error message. It is unclear to me if `Pretype_errors.PretypeError _` is a legit exception that can be raised by tactics.